weatherrouting/routers/linearbestisorouter.py

```diff
@@ -26,26 +26,43 @@ class LinearBestIsoRouter (Router):
 	}
 
 	def route (self, lastlog, time, start, end):
-		if lastlog != None and len (lastlog.isochrones) > 0:
-			isoc = self.calculateIsochrones (time + datetime.timedelta(hours=1), lastlog.isochrones, end)
-		else:
-			isoc = self.calculateIsochrones (time + datetime.timedelta(hours=1), [[(start[0], start[1], time)]], end)
 
 		position = start
 		path = []
-		for p in isoc[-1]:
-			distance_to_end_point = utils.pointDistance (end[0],end[1], p[0], p[1])
-			if distance_to_end_point < self.getParamValue('minIncrease'):
-				(twd,tws) = self.grib.getWindAt (time + datetime.timedelta(hours=1), p[0], p[1])
-				maxReachDistance = self.polar.maxReachDistance(p, twd, tws)
-				if utils.pointDistance (end[0],end[1], p[0], p[1]) < abs(maxReachDistance*1.1):
-					if (not self.pointValidity or self.pointValidity(end[0],end[1])) and (not self.lineValidity or self.lineValidity(end[0],end[1], p[0], p[1])):
-						path.append (p)
-						for iso in isoc[::-1][1::]:
-							path.append (iso[path[-1][2]])
-
-						path = path[::-1]
-						position = path[-1]
-						break
-
-		return RoutingResult(time=time + datetime.timedelta(hours=1), path=path, position=position, isochrones=isoc)
+
+		def generate_path(p):
+			nonlocal path
+			nonlocal isoc
+			nonlocal position
+			path.append (p)
+			for iso in isoc[::-1][1::]:
+				path.append (iso[path[-1][2]])
+			path = path[::-1]
+			position = path[-1]
+		
+		if self.grib.getWindAt (time + datetime.timedelta(hours=1), end[0],end[1]):
+			if lastlog != None and len (lastlog.isochrones) > 0:
+				isoc = self.calculateIsochrones (time + datetime.timedelta(hours=1), lastlog.isochrones, end)
+			else:
+				isoc = self.calculateIsochrones (time + datetime.timedelta(hours=1), [[(start[0], start[1], time)]], end)
+			for p in isoc[-1]:
+				distance_to_end_point = utils.pointDistance (end[0],end[1], p[0], p[1])
+				if distance_to_end_point < self.getParamValue('minIncrease'):
+					(twd,tws) = self.grib.getWindAt (time + datetime.timedelta(hours=1), p[0], p[1])
+					maxReachDistance = self.polar.maxReachDistance(p, twd, tws)
+					if utils.pointDistance (end[0],end[1], p[0], p[1]) < abs(maxReachDistance*1.1):
+						if (not self.pointValidity or self.pointValidity(end[0],end[1])) and (not self.lineValidity or self.lineValidity(end[0],end[1], p[0], p[1])):
+							generate_path(p)
+							break
+						
+		else: #out of grib scope
```

Comment:
The PR suggests a second exit condition: when algorithm goes out of grib temporal scope (getWindAt grib class method returns None) it rescans last calculated isochrones to find the nearest to end point of track and generate a partial solution with new generate_path function (created only to avoid code duplication...)

Reply:
seems legit